src/mbgl/map/map_impl.cpp

```diff
@@ -172,8 +172,19 @@ void Map::Impl::jumpTo(const CameraOptions& camera) {
     onUpdate();
 }
 
-void Map::Impl::onStyleImageMissing(const std::string& id) {
-    observer.onStyleImageMissing(id);
+void Map::Impl::onStyleImageMissing(const std::string& id, std::function<void()> done) {
+
+    if (style->getImage(id) != nullptr) {
+        done();
+        return;
+    }
+
+    observer.onStyleImageMissing(id, [this, done](optional<std::unique_ptr<mbgl::style::Image>> image = {}) {
```

Comment:
Can this lambda be called after `Map::Impl` object is deleted?

Reply:
This callback was removed in a later commit.